program/src/lib.rs

```diff
@@ -1,4 +1,5 @@
 #![allow(clippy::arithmetic_side_effects)]
+#![allow(deprecated)]
```

Comment:
This was needed to keep clippy happy without having to update the code – not sure whether it would be better the update the code or not.

Reply:
Agree, looks great, much better to update the code.

Reply:
Ok great! Can you push the changes to your branch? Then I can approve and we can merge